### ISSUE TITLE
Add toggle for additional settings

### ISSUE THREAD

**Issue by mclindy** [Atheos/Atheos#280]
### Feature Request and/or another set of eyes request

I'd like to add toggles in Settings->Editor Settings to enable/disable additional options, such as behavioursEnabled, enableBAsicAutocompletion, enableLiveAutocompletion, enableSnippets, etc. 

I have made some edits trying to handle this myself, but I think I'm missing something obvious.

If anyone can point me in the right direction I'd appreciate it. 

Below are some related screenshots of the changes I have made to a development instance.

First, I edited **components/settings/panels/editor.php** to add to the settings panel. I inserted code at lines 69 - 79:
![componentsSettingsPanels](https://github.com/Atheos/Atheos/assets/16168011/f4f04008-13b8-427a-90c8-d3cf57b360b7)

And here's how it looks in the IDE. I had just toggled the setting and you'll notice the settings saved message in the lower-right corner.
![settingsDisabledAndSaved](https://github.com/Atheos/Atheos/assets/16168011/efbdd0c1-5b6a-4238-a3ef-7a7b6b87499f)

Here's a screenshot of **the user's settings.db.json**, where you can see the the toggled setting has been saved:
![settingsStoredForUser](https://github.com/Atheos/Atheos/assets/16168011/e5da7e67-a2ea-46e2-8e77-21d8503b5f77)

Here's a change I made to **components/editor/init.js**, where I'm attempting to load the user's preferences using the other parameters as a reference. In **lines 132 and 133** you can see that I tried two different ways of achieving this, but haven't had success yet. 
![componentsEditorInitJS](https://github.com/Atheos/Atheos/assets/16168011/cde26cdd-48de-49ae-813b-56bcc26bb3f4)

Here's another addition to **components/editor/init.js**, where I added lines 400-425 hoping to add a function/method using surrounding functions as kind of template. I tried it two different ways with no love (one is commented out).
![componentsEditorInitJS400](https://github.com/Atheos/Atheos/assets/16168011/e5048c88-3269-458c-ab61-a08add1d7824)

Here, I added lines 134-136 to the switch in components/settings/init.js:
![componentsSettingsInitJS](https://github.com/Atheos/Atheos/assets/16168011/4b43ea96-2c03-46d4-84cc-ec6300d7f60c)

I think I'm close, but I am clearly missing something and/or I've got a spelling error or missing semi-colon type of error somewhere. Any help would be appreciated. Atheos has helped me in the classroom a lot, and I was hoping to be able to give back a little and not have to rely on you all to do it for me, but this is as far as I have gotten. I'll keep plugging away in the meantime.

Thanks,
Greg

**Comment by mclindy** [comment]
Cleaned up some grammar.

**Comment by HLSiira** [comment]
I pretty much had to implement the entire thing to recall how this area worked lol. The Ace Editor is weird, some things are instances, some things are editors, and some settings need to be passed as objects, some have functions that can be poked. It's a lot of guess and check with the development tools.

**Comment by mclindy** [comment]
Thank you for the prompt action! Reviewing your changes helps me learn about where I fell short, too. I haven't had a chance to implement this with my classes yet, but I will this week. It's going to make an important difference. For example, I turn off tag completion at the system level because I want beginners to have to do it manually to build the habit. At the same time, I have a few special needs kids (e.g. dyslexia) that greatly benefit from having some features turned on. In the past I have made a second instance of Atheos, changed the settings, and had the student log into that instance. It works but would be hard to keep track of and is also prob a security risk on my end. So again, thank you. This is a very nice change for classroom use.

**Comment by mclindy** [comment]
I had time to pull into a new instance and this works beautifully. Thank you!

I see a new bug that is kind of important, however, and I hope it's not related to this feature request.

On a new installation, when I create the initial (admin) account I get a popup that says "An error has occurred" or something vague like that. Then I need to confirm by clicking create account again. Instead of logging me in and taking me to the admins screen I am taken to the login screen. You made some security updates, and maybe having to log in is part of that update.

However, the bug is that even though I left development mode enabled I am unable to edit Atheos code as before, and the </> icon is missing from the projects area.

**TLDR: The bug is that when creating the first user in a new installation, an error is thrown and you are unable to enable development mode.**

Again thanks for the fix. I hate to complain, but I'd rather you hear it quickly from me than find out later from people trying to install it for the first time. 

Great work.
-Greg

**Comment by mclindy** [comment]
In order to resolve the icon issue a slight change in **/public/fonts.min.css** is needed:
`.html5-icon:before{font-family:'Font Awesome 5 Brands'; ... }`
change to
`'Font Awesome 6 brands';`
_Note the lower case!_
Do the same for **.css3-icon.**

**Comment by mclindy** [comment]
Thanks! Hey, Atheos is awesome. I'm glad you take the time to keep developing.
